modules/light-clients/06-solomachine/light_client_module_test.go

```diff
@@ -1,21 +1,130 @@
 package solomachine_test
 
 import (
-	fmt "fmt"
+	"fmt"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v8/modules/core/23-commitment/types"
 	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+	ibcerrors "github.com/cosmos/ibc-go/v8/modules/core/errors"
 	"github.com/cosmos/ibc-go/v8/modules/core/exported"
 	solomachine "github.com/cosmos/ibc-go/v8/modules/light-clients/06-solomachine"
 	ibctm "github.com/cosmos/ibc-go/v8/modules/light-clients/07-tendermint"
 	ibctesting "github.com/cosmos/ibc-go/v8/testing"
+	ibcmock "github.com/cosmos/ibc-go/v8/testing/mock"
 )
 
 const (
-	smClientID   = "06-solomachine-100"
-	wasmClientID = "08-wasm-0"
+	smClientID       = "06-solomachine-100"
+	unusedSmClientID = "06-solomachine-999"
+	wasmClientID     = "08-wasm-0"
 )
 
+func (suite *SoloMachineTestSuite) TestStatus() {
+	var (
+		clientState *solomachine.ClientState
+		clientID    string
+	)
+
+	testCases := []struct {
+		name      string
+		malleate  func()
+		expStatus exported.Status
+	}{
+		{
+			"client is active",
+			func() {},
+			exported.Active,
+		},
+		{
+			"client is frozen",
+			func() {
+				clientState = solomachine.NewClientState(0, &solomachine.ConsensusState{})
+				clientState.IsFrozen = true
+				suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientState(suite.chainA.GetContext(), clientID, clientState)
+			},
+			exported.Frozen,
+		},
+		{
+			"cannot find subject client state",
+			func() {
+				clientID = unusedSmClientID
+			},
+			exported.Unknown,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			clientID = suite.solomachine.ClientID
+
+			lightClientModule, found := suite.chainA.App.GetIBCKeeper().ClientKeeper.Route(clientID)
+			suite.Require().True(found)
+
+			suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientState(suite.chainA.GetContext(), clientID, suite.solomachine.ClientState())
+
+			tc.malleate()
+
+			status := lightClientModule.Status(suite.chainA.GetContext(), clientID)
+			suite.Require().Equal(tc.expStatus, status)
+		})
+	}
+}
+
+func (suite *SoloMachineTestSuite) TestGetTimestampAtHeight() {
+	var clientID string
+	height := clienttypes.NewHeight(0, suite.solomachine.ClientState().Sequence)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expValue uint64
+		expErr   error
+	}{
+		{
+			"success: get timestamp at height exists",
+			func() {},
+			suite.solomachine.ClientState().ConsensusState.Timestamp,
+			nil,
+		},
+		{
+			"failure: client not found",
```

Comment:
Just for consistency:

```suggestion
			"failure: cannot find client state",
```